homeassistant/components/google/__init__.py

```diff
@@ -220,11 +220,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Google from a config entry."""
     hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {}
+
     implementation = (
         await config_entry_oauth2_flow.async_get_config_entry_implementation(
             hass, entry
         )
     )
+    if not isinstance(
+        implementation, config_entry_oauth2_flow.LocalOAuth2Implementation
+    ):
+        raise ValueError(f"Unexpected auth implementation {implementation}")
+    if entry.unique_id is None:
+        hass.config_entries.async_update_entry(
+            entry, unique_id=implementation.client_id
```

Comment:
Shouldn't we allow a config entry per user that authenticates with an application? Many users can authenticate with the same application, right?

Reply:
Good point.  I had in mind the nest integration where you make a device access project tied to a specific account. (Also, we may be able to make nabu casa a partner for this some day, so it could support more than local oauth anyway.)

In general, you don't get back unique identifiers from Google's OAuth API. However, we do have a hack that the primary google calendar is your email address and we can use that, which we set as the [title in the config flow](https://github.com/home-assistant/core/blob/ab95299150ea489d42b6903454b9e064344f98fb/homeassistant/components/google/config_flow.py#L152) -- however, it's a little more heavy weight as it requires another API call (e.g. won't work if reauth is needed)

Is using the primary calendar id the way to go?

Reply:
Primary calendar id sounds good.

Reply:
Implemented. Thanks for flagging that, it works great! I did testing for a few different scenarios and it all looks good and seems to work well.